File: src/python/_0643_Maximum_Average_Subarray_I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currSum = 0
        maxSum = float("-inf")

        for i, num in enumerate(nums):
            currSum += nums[i]
            if i >= k:
                currSum -= nums[i - k]
            if i >= k - 1:
                maxSum = max(maxSum, currSum)

        return maxSum / k

File: src/_0643_Maximum_Average_Subarray_I.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int currSum = 0, maxSum = Integer.MIN_VALUE;

        for (int i = 0; i < nums.length; i++) {
            currSum += nums[i];
            if (i >= k)
                currSum -= nums[i - k];
            if (i >= k - 1)
                maxSum = maxSum > currSum ? maxSum : currSum;
        }

        return (double) maxSum / k;
    }
}
